app/models/current_user_data.rb

```diff
@@ -0,0 +1,43 @@
+CurrentUserData = Struct.new(:user, :restricted) do
+  def data
+    serializer.new(user, scope: serializer_scope, root: :current_user).as_json.tap do |json|
+      json[:current_user].except!(:group_ids, :membership_ids)
+    end
+  end
+
+  private
+
+  def serializer
+    if restricted
+      Restricted::UserSerializer
+    else
+      Full::UserSerializer
+    end
+  end
+
+  def serializer_scope
+    { memberships: memberships }.tap do |hash|
+      hash.merge!(
```

Comment:
Always serialize out memberships, because restricted users need em and logged out users don't have any.